src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,51 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (checkLogin(user.getLogin())
+                && checkPassword(user.getPassword())
+                && checkAge(user.getAge())) {
+            storageDao.add(user);
+        }
+        return user;
+    }
+
+    public boolean checkLogin(String login) {
+        if (login == null) {
+            throw new CustomException(
+                    "Login can't be empty, login must be at list 6 characters long"
+            );
+        }
+        if (storageDao.get(login) != null) {
+            throw new CustomException("User with such login is already registered");
+        }
+        if (login.length() < 6) {
+            throw new CustomException("User login must be at least 6 characters long");
+        }
+        return true;
+    }
+
+    public boolean checkPassword(String password) {
+        boolean passwordIsOk = true;
+        if (password == null) {
+            passwordIsOk = false;
+            throw new CustomException("Password can't be empty");
+        }
+        if (password.length() < 6) {
+            passwordIsOk = false;
+            throw new CustomException("Password must be at least 6 characters long");
+        }
+        return passwordIsOk;
```

Comment:
```suggestion
        if (password == null) {
            throw new CustomException("Password can't be empty");
        }
        if (password.length() < REQUIRED_PASSWORD_LENGTH) {
            throw new CustomException("Password must be at least 6 characters long");
        }
        return true;
```
your exception is runtime exception, it exits from the method, no need to set false